pkg/trace/api/profiles.go

```diff
@@ -76,6 +105,56 @@ func newProfileProxy(transport http.RoundTripper, target *url.URL, apiKey, tags
 	return &httputil.ReverseProxy{
 		Director:  director,
 		ErrorLog:  stdlog.New(logger, "profiling.Proxy: ", 0),
-		Transport: transport,
+		Transport: &multiTransport{transport, targets, keys},
+	}
+}
+
+// multiTransport sends HTTP requests to multiple targets using an
+// underlying http.RoundTripper. API keys are set separately for each target.
+// When multiple endpoints are in use the response from the main endpoint
+// is proxied back to the client, while for all aditional endpoints the
+// response is discarded. There is no de-duplication done between endpoint
+// hosts or api keys.
+type multiTransport struct {
+	rt      http.RoundTripper
+	targets []*url.URL
+	keys    []string
+}
+
+func (m *multiTransport) RoundTrip(req *http.Request) (*http.Response, error) {
+	setTarget := func(r *http.Request, u *url.URL, apiKey string) {
+		r.Host = u.Host
+		r.URL = u
+		r.Header.Set("DD-API-KEY", apiKey)
+	}
+	if len(m.targets) == 1 {
+		setTarget(req, m.targets[0], m.keys[0])
+		return m.rt.RoundTrip(req)
+	}
+	slurp, err := ioutil.ReadAll(req.Body)
+	if err != nil {
+		return nil, err
+	}
+	var (
+		resp *http.Response
+		rerr error
+	)
+	for i, u := range m.targets {
+		newreq := req.Clone(req.Context())
+		newreq.Body = ioutil.NopCloser(bytes.NewReader(slurp))
+		setTarget(newreq, u, m.keys[i])
+		if i == 0 {
+			// given the way we construct the list of targets the main endpoint
+			// will be the first one called, we return its response and error
+			resp, rerr = m.rt.RoundTrip(newreq)
+			err = rerr
+		} else {
+			// we discard responses for all subsequent requests and log all errors
+			_, err = m.rt.RoundTrip(newreq)
```

Comment:
We might have a leak here if we discard this response body. If the `err == nil` someone has to close it according to [documentation](https://golang.org/pkg/net/http/#Client.Do): 

> If the returned error is nil, the Response will contain a non-nil Body which the user is expected to close. If the Body is not both read to EOF and closed, the Client's underlying RoundTripper (typically Transport) may not be able to re-use a persistent TCP connection to the server for a subsequent "keep-alive" request.

The `http.ReverseProxy` closes any returned response, so we're good for the first one, but subsequent ones are on us, so we'll have to do something like:

```go
io.Copy(ioutil.Discard, resp.Body)
resp.Body.Close()
```

Reply:
Good catch, thanks! 🙌 I don't know why I assumed I wouldn't need to manage resources for a discarded variable 😃

Reply:
Honestly it isn't very obvious. It's all good. Very common mistake.